v2/car.go

```diff
@@ -0,0 +1,70 @@
+package car
```

Comment:
is the plan to do a v2 dir inside the Go v1 module, or to do an entirely separate Go v2 module? assuming we want the latter, I think it would be best to target PRs to e.g. a wip-v2 branch, so we can gradually review+merge and track progress that way without messing with master or having an ever-growing PR.

Reply:
That makes sense; I'll reopen the PR agains a v2 branch. 

Re v2 module I share the same thoughts. This PR indeed defines a new v2 module

Thanks @mvdan

Reply:
this seems like a reasonable direction